src/app/server/JointFabricDatastore.cpp

```diff
@@ -518,25 +670,29 @@ JointFabricDatastore::AddBindingToEndpointForNode(
     {
         if (entry.nodeID == nodeId && entry.endpointID == endpointId)
         {
-            // TODO: mark as pending, update device, then mark as committed
-            entry.binding = binding;
-            return CHIP_NO_ERROR;
+            if (BindingMatches(entry.binding, binding))
+            {
+                return CHIP_NO_ERROR;
+            }
         }
     }
 
     VerifyOrReturnError(mEndpointBindingEntries.size() < kMaxGroups, CHIP_ERROR_NO_MEMORY);
 
     // Create a new binding entry
     Clusters::JointFabricDatastore::Structs::DatastoreEndpointBindingEntryStruct::Type newBindingEntry;
-    newBindingEntry.nodeID            = nodeId;
-    newBindingEntry.endpointID        = endpointId;
-    newBindingEntry.binding           = binding;
-    newBindingEntry.statusEntry.state = Clusters::JointFabricDatastore::DatastoreStateEnum::kCommitted;
+    newBindingEntry.nodeID     = nodeId;
+    newBindingEntry.endpointID = endpointId;
+    newBindingEntry.binding    = binding;
+    ReturnErrorOnFailure(GenerateAndAssignAUniqueListID(newBindingEntry.listID));
+    newBindingEntry.statusEntry.state = Clusters::JointFabricDatastore::DatastoreStateEnum::kPending;
+
+    // TODO: Update device
```

Comment:
TODO probably is most appropriate to have after adding the binding to the datastore (2 lines below)

Reply:
Fixed in latest commit.